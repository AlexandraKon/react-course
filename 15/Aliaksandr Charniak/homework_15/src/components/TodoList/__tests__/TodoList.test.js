import React from 'react'
import { configure, shallow } from 'enzyme'
import Adapter from '@wojtekmaj/enzyme-adapter-react-17'

import TodoList from '..'
import Todo from '../../Todo'

configure({ adapter: new Adapter() })

describe('<TodoList /> mounting behaviour', () => {
  const mockFn = jest.fn()

  const props = {
    toggleTodo: mockFn,
    todos: [{ id: 0, completed: false, text: 'Some text...' }],
  }

  let wrapper

  beforeEach(() => {
    wrapper = shallow(<TodoList {...props} />)
  })

  it('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot()
  })

  it('should render 1 <TodoList /> component', () => {
    expect(wrapper).toHaveLength(1)
  })

  it('should correctly render children', () => {
    expect(wrapper.containsMatchingElement(<Todo />)).toEqual(true)
  })

  it('should correctly count children', () => {
    expect(wrapper.find('Todo')).toHaveLength(1)
  })
})
