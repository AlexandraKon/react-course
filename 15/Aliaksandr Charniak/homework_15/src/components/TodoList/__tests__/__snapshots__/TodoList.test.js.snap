// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TodoList /> mounting behaviour should match snapshot 1`] = `
<ul>
  <Todo
    completed={false}
    id={0}
    key="0"
    onClick={[Function]}
    text="Some text..."
  />
</ul>
`;
